File: backend/src/main/java/com/interview/platform/mod03_department_management/repository/DepartmentRepository.java
package com.interview.platform.mod03_department_management.repository;

import com.interview.platform.mod03_department_management.model.DepartmentEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.UUID;

public interface DepartmentRepository extends JpaRepository<DepartmentEntity, UUID> {
}

File: android-app/app/src/main/java/com/interview/platform/domain/repository/DepartmentRepository.kt
package com.interview.platform.domain.repository

import com.interview.platform.domain.model.mod03_department_management.Department
import com.interview.platform.data.model.Result
import kotlinx.coroutines.flow.Flow

interface DepartmentRepository {
    fun fetchAll(): Flow<Result<List<Department>>>
    suspend fun syncData(): Result<Unit>
}
